localstack/services/s3/notifications.py

```diff
@@ -1,3 +1,5 @@
+from __future__ import annotations
+
```

Comment:
Because of the conditional imports of Moto classes, with it the types evaluation are postponed and will not fail. It's not great, but the other solution would be to split out the notifications into Moto/no Moto, or import inside the context creation, which is maybe better but incurs a little performance cost.